src/exposition/http.rs

```diff
@@ -293,59 +277,27 @@ mod handlers {
                 continue;
             }
 
-            if let Some(_counter) = any.downcast_ref::<Counter>() {
-                metadata.counters.push(MetricMetadata {
-                    name: metric.formatted(metriken::Format::Simple),
-                });
-            } else if let Some(_gauge) = any.downcast_ref::<Gauge>() {
-                metadata.gauges.push(MetricMetadata {
-                    name: metric.formatted(metriken::Format::Simple),
-                });
-            } else if let Some(histogram) = any.downcast_ref::<RwLockHistogram>() {
-                metadata.histograms.push((
+            if let Some(counter) = any.downcast_ref::<Counter>() {
+                readings.counters.push((
                     MetricMetadata {
                         name: metric.formatted(metriken::Format::Simple),
                     },
-                    histogram.config(),
+                    counter.value(),
```

Comment:
Instead of doing downcasting here you may want to use [`Metric::value`](https://docs.rs/metriken/latest/metriken/trait.Metric.html#tymethod.value) instead. You will still need to downcast for the histograms, but counters and gauges are supported.